### ISSUE TITLE
`incompatible_msrv` thinks `.await` is stabilized since 1.64 while it's not 

### ISSUE THREAD

**Issue by oxalica** [rust-lang/rust-clippy#12273]
### Summary

When MSRV is set to <1.64, eg. `1.63`, clippy nightly warns on every `.await` and thinks it's stable since 1.64. But actually it compiles fine on rustc since 1.39.

### Lint Name

incompatible_msrv

### Reproducer

Given `lib.rs`:

```rust
pub async fn foo<Fut: std::future::Future<Output = ()>>(fut: Fut) {
    fut.await;
}
```

and `Cargo.toml`:

```toml
[package]
name = "testt"
version = "0.1.0"
edition = "2018"
rust-version = "1.63"
```

Run `cargo clippy` on nightly.
```
warning: current MSRV (Minimum Supported Rust Version) is `1.63.0` but this item is stable since `1.64.0`
 --> src/lib.rs:2:9
  |
2 |     fut.await;
  |         ^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#incompatible_msrv
  = note: `#[warn(clippy::incompatible_msrv)]` on by default

warning: `testt` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
```

I expected to see this happen: no warnings. Because the code actually compiles on rustc 1.63.

### Version

```text
rustc 1.78.0-nightly (6cc484351 2024-02-10)
binary: rustc
commit-hash: 6cc4843512d613f51ec81aba689180c31b0b28b6
commit-date: 2024-02-10
host: x86_64-unknown-linux-gnu
release: 1.78.0-nightly
LLVM version: 17.0.6
```
